### DISCUSSION TITLE
Running efficiency while setting requires_grad=False for both input and output embeddings of LLM

### DISCUSSION THREAD

**Discussion by VERSPD0** [haotian-liu/LLaVA#374] [General]
Hi, thanks for your work!

While going through the code, I noticed the way to froze the LLM is to setting `requires_grad=False` for both input and output embeddings. Though I do understand this implementation can block the backpropagation for those parameters between the input and output layers, I still concerned about whether it would make it slow during training because gradients of parameters with `requires_grad=True` in the LLM may still be calculated.
